verify-sna/functions/check-verify.js

```diff
@@ -1,79 +1,243 @@
+                    WHERE phone_number = ? AND sna_url != ?;
+                    `,
+                ['expired', check.to, sortedRows[0].sna_url],
+                (err) => {
+                  if (err) {
+                    const statusCode = err.status || 400;
+                    response.setStatusCode(statusCode);
+                    response.setBody({
+                      message: err.message,
+                    });
+                    return callback(null, response);
+                  }
+                  return callback(null, response);
+                }
+              );
+            }
+          );
+        } else {
+          return callback(null, response);
+        }
+      } else {
+        db.run(
+          `
+            INSERT INTO verifications (phone_number, sna_url, status, verification_start_datetime, verification_check_datetime)
+            VALUES (?, ?, ?, NULL, DATETIME('NOW'));
+            `,
+          [check.to, 'no sna url generated', response.body.status],
+          (err) => {
+            if (err) {
+              const statusCode = err.status || 400;
+              response.setStatusCode(statusCode);
+              response.setBody({
+                message: err.message,
+              });
+              return callback(null, response);
+            }
+            return callback(null, response);
+          }
+        );
+      }
+    }
+  );
+}
 
+// eslint-disable-next-line consistent-return
 exports.handler = async function (context, event, callback) {
   const response = new Twilio.Response();
   response.appendHeader('Content-Type', 'application/json');
 
   try {
     const client = context.getTwilioClient();
     const service = context.VERIFY_SERVICE_SID;
-    const rabbitmqUrl = context.RABBIT_MQ_URL;
 
     const [countryCode, phoneNumber] = [event.countryCode, event.phoneNumber];
 
-    // TODO: Check that country code and phone number are present and correct
-
-    /**
-     * const check = await client.verify
-     * .services(service)
-     * .verificationChecks
-     * .create({to: countryCode + phoneNumber});
-     */
-
-    const check = {
-      status: 'approved',
-    };
+    const check = await client.verify.v2
+      .services(service)
+      .verificationChecks.create({ to: `${countryCode}${phoneNumber}` });
 
     if (check.status === 'approved') {
       response.setStatusCode(200);
       response.setBody({
-        message: 'Phone number verified successfully',
+        status: 'verified',
+        message: 'SNA verification successful, phone number verified',
       });
+    } else {
+      if (
+        check.snaAttemptsErrorCodes[check.snaAttemptsErrorCodes.length - 1]
+          .code === 60519 ||
+        check.snaAttemptsErrorCodes[check.snaAttemptsErrorCodes.length - 1]
+          .code > 60520
+      ) {
+        response.setStatusCode(200);
+        response.setBody({
+          status: 'pending',
```

Comment:
I was in fact returning a response that was true or false. However, while going through the API documentation I noticed that it can't just be a boolean value because there are three possible cases:
1. When the verification is successful
2. When the verification is unsuccessful
3. When the verification hasn't still been verified. In this last case, I don't think it is right to return an unsuccessful state because it can potentially lead to a successful state.
That's the explanation of why I changed the response we had agreed on. Let me know if I'm mistaken about that.

Reply:
I understand the diverse responses of the Verify API, however, for the sake of this pilot, we are just interested on whether the verification was or not successful. My proposal:

Whenever we get to the use case #3, let's convert this into a 400, and add a text to the message variable indicating this situation. If you think about it, this will happen whenever you call this endpoint without calling the EVURL, after a wrong EVURL response or even starting a new verification, which makes no sense in the context of the workflow.